storage/SMAIndexSubBlock.hpp

```diff
@@ -0,0 +1,427 @@
+/**
+ *   Copyright 2016 Pivotal Software, Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ **/
+
+#ifndef QUICKSTEP_STORAGE_SMA_INDEX_SUB_BLOCK_HPP_
+#define QUICKSTEP_STORAGE_SMA_INDEX_SUB_BLOCK_HPP_
+
+#include <stdint.h>
+#include <cstddef>
+#include <iostream>
+
+#include "catalog/CatalogTypedefs.hpp"
+#include "expressions/predicate/PredicateCost.hpp"
+#include "storage/IndexSubBlock.hpp"
+#include "storage/StorageBlockInfo.hpp"
+#include "storage/SubBlockTypeRegistryMacros.hpp"
+#include "types/TypeID.hpp"
+#include "types/TypedValue.hpp"
+#include "types/operations/binary_operations/BinaryOperation.hpp"
+#include "types/operations/comparisons/Comparison.hpp"
+#include "types/operations/comparisons/ComparisonFactory.hpp"
+#include "types/operations/comparisons/ComparisonID.hpp"
+#include "utility/Macros.hpp"
+#include "utility/PtrVector.hpp"
+
+#include "glog/logging.h"
+
+namespace quickstep {
+
+class SMAIndexSubBlock;
+
+QUICKSTEP_DECLARE_SUB_BLOCK_TYPE_REGISTERED(SMAIndexSubBlock);
+
+/**
+ * Namespace contains helper functions of the SMA block. This includes functions
```

Comment:
As I believe this is the first place to have `SMA`, I suggest to have the full version of the phrase, instead of the shortened one.